.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -99,29 +103,46 @@ package_dependencies_arm64:
     ARCHIVE_NAME: dependencies-arm64.tar.gz
     ARCH: arm64
 
+.kernel_matrix_testing_new_profile:
+  - mkdir -p ~/.aws
+  - set +x
+  - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.agent-qa-profile --with-decryption --query "Parameter.Value" --out text >> ~/.aws/config
+  - set -x
+  - export AWS_PROFILE=agent-qa-ci
 
 kernel_matrix_testing_setup_env:
   extends:
     - .kitchen_ec2_location_us_east_1
   stage: kernel_matrix_testing
   rules:
     !reference [.manual]
-  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner:7c39524b
   needs: ["go_deps", "go_tools_deps"]
-  tags: ["runner:main"]
+  tags: ["arch:amd64"]
   variables:
     EC2_X86_INSTANCE_TYPE: "i3.metal"
     EC2_ARM_INSTANCE_TYPE: "m6g.metal"
-    X86_AMI_ID: "ami-0f50812e4388946ff"
-    ARM_AMI_ID: "ami-0d0b8e440bb126e28"
+    X86_AMI_ID: "ami-0b8f21d0afb2fa930"
+    ARM_AMI_ID: "ami-097810db042c1c0d8"
     AWS_REGION: us-east-1
     LibvirtSSHKeyX86: $CI_PROJECT_DIR/libvirt_rsa-x86
     LibvirtSSHKeyARM: $CI_PROJECT_DIR/libvirt_rsa-arm
     STACK_DIR: $CI_PROJECT_DIR/stack.dir
+    # The ssh key is created by the pulumi scenario, to be used for creating
+    # instances in the build-stable account. We reuse this file to ssh into
+    # the instances in subsequent jobs.
+    AWS_EC2_SSH_KEY_FILE: $CI_PROJECT_DIR/ssh_key
+    AWS_EC2_SSH_KEY_NAME: datadog-agent-ci
+    INFRA_ENV: "aws/agent-qa"
+  before_script:
+    - !reference [.retrieve_linux_go_deps]
+    - !reference [.kernel_matrix_testing_new_profile]
   script:
-    - echo s3://dd-ci-datadog-agent-omnibus-cache-build-stable/kernel-matrix-testing/$CI_PIPELINE_ID/$CI_JOB_ID/pulumi > $STACK_DIR
+    - echo "s3://dd-pulumi-state?region=us-east-1&awssdk=v2&profile=$AWS_PROFILE" > $STACK_DIR
     - pulumi login $(cat $STACK_DIR | tr -d '\n')
-    - inv -e system-probe.test-microvms --security-groups=$KITCHEN_EC2_SG_IDS --subnets=$KITCHEN_EC2_SUBNET --instance-type-x86=$EC2_X86_INSTANCE_TYPE --instance-type-arm=$EC2_ARM_INSTANCE_TYPE --x86-ami-id=$X86_AMI_ID --arm-ami-id=$ARM_AMI_ID
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.ssh_key --with-decryption --query "Parameter.Value" --out text > $AWS_EC2_SSH_KEY_FILE
```

Comment:
Does this need to be wrapped with `set +x/set -x`?

Reply:
I do not think so. Firstly the output is being redirected and it does not get printed.
Secondly this same pattern is used elsewhere as well for example [this](https://github.com/DataDog/datadog-agent/blob/main/.gitlab/functional_test/system_probe.yml#L22) job call run-test-kitchen which does the [same](https://github.com/DataDog/datadog-agent/blob/main/test/kitchen/tasks/run-test-kitchen.sh#L98) thing.

Reply:
Alternatively this job does what you suggest: https://github.com/DataDog/datadog-agent/blob/main/.gitlab/e2e.yml#L131

I will move to this approach too.